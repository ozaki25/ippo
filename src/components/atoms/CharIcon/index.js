import React from 'react';
import { Avatar } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import {
  red,
  pink,
  purple,
  deepPurple,
  indigo,
  blue,
  lightBlue,
  cyan,
  teal,
  green,
  lightGreen,
  lime,
  yellow,
  amber,
  orange,
  deepOrange,
  brown,
  grey,
  blueGrey,
} from '@material-ui/core/colors/';
import classnames from 'classnames';
import propTypes from 'prop-types';

const styles = {
  avatar: {
    fontSize: '24px',
    height: '48px',
    width: '48px',
  },
  red: {
    backgroundColor: red[500],
  },
  pink: {
    backgroundColor: pink[500],
  },
  purple: {
    backgroundColor: purple[500],
  },
  deepPurple: {
    backgroundColor: deepPurple[500],
  },
  indigo: {
    backgroundColor: indigo[500],
  },
  blue: {
    backgroundColor: blue[500],
  },
  lightBlue: {
    backgroundColor: lightBlue[500],
  },
  cyan: {
    backgroundColor: cyan[500],
  },
  teal: {
    backgroundColor: teal[500],
  },
  green: {
    backgroundColor: green[500],
  },
  lightGreen: {
    backgroundColor: lightGreen[500],
  },
  lime: {
    backgroundColor: lime[500],
  },
  yellow: {
    backgroundColor: yellow[500],
  },
  amber: {
    backgroundColor: amber[500],
  },
  orange: {
    backgroundColor: orange[500],
  },
  deepOrange: {
    backgroundColor: deepOrange[500],
  },
  brown: {
    backgroundColor: brown[500],
  },
  grey: {
    backgroundColor: grey[500],
  },
  blueGrey: {
    backgroundColor: blueGrey[500],
  },
};

const colorMap = {
  0: 'red',
  1: 'pink',
  2: 'purple',
  3: 'deepPurple',
  4: 'indigo',
  5: 'blue',
  6: 'lightBlue',
  7: 'cyan',
  8: 'teal',
  9: 'green',
  10: 'lightGreen',
  11: 'lime',
  12: 'yellow',
  13: 'amber',
  14: 'orange',
  15: 'deepOrange',
  16: 'brown',
  17: 'grey',
  18: 'blueGrey',
};

const CharIcon = ({ name, classes }) => {
  const size = Object.keys(colorMap).length;
  const code = name.charCodeAt();
  const color = colorMap[code % size];
  return <Avatar className={classnames(classes.avatar, classes[color])}>{name.charAt(0)}</Avatar>;
};

CharIcon.displayName = 'CharIcon';

CharIcon.propTypes = {
  name: propTypes.string.isRequired,
};

export default withStyles(styles)(CharIcon);
