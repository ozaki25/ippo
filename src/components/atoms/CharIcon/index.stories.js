import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import CharIcon from '.';

const stories = storiesOf('atoms/CharIcon', module);

const props = {};

stories.add('英語小文字', () => <CharIcon {...props} name={text('name', 'ozaki25')} />);

stories.add('英語大文字', () => <CharIcon {...props} name={text('name', 'Ozaki')} />);

stories.add('漢字', () => <CharIcon {...props} name={text('name', '尾崎')} />);

stories.add('カタカナ', () => <CharIcon {...props} name={text('name', 'オザキ')} />);
