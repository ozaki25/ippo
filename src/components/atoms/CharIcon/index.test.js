import React from 'react';
import { snapshot } from 'test/helpers';
import CharIcon from '.';

const props = {
  name: 'ozaki25',
};

snapshot('CharIcon/nomal', <CharIcon {...props} />);
