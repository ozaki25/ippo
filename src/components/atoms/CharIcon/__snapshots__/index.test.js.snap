// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: CharIcon/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <WithStyles(CharIcon)
    name="ozaki25"
  >
    <CharIcon
      classes={
        Object {
          "amber": "CharIcon-amber-15",
          "avatar": "CharIcon-avatar-1",
          "blue": "CharIcon-blue-7",
          "blueGrey": "CharIcon-blueGrey-20",
          "brown": "CharIcon-brown-18",
          "cyan": "CharIcon-cyan-9",
          "deepOrange": "CharIcon-deepOrange-17",
          "deepPurple": "CharIcon-deepPurple-5",
          "green": "CharIcon-green-11",
          "grey": "CharIcon-grey-19",
          "indigo": "CharIcon-indigo-6",
          "lightBlue": "CharIcon-lightBlue-8",
          "lightGreen": "CharIcon-lightGreen-12",
          "lime": "CharIcon-lime-13",
          "orange": "CharIcon-orange-16",
          "pink": "CharIcon-pink-3",
          "purple": "CharIcon-purple-4",
          "red": "CharIcon-red-2",
          "teal": "CharIcon-teal-10",
          "yellow": "CharIcon-yellow-14",
        }
      }
      name="ozaki25"
    >
      <WithStyles(Avatar)
        className="CharIcon-avatar-1 CharIcon-brown-18"
      >
        <Avatar
          className="CharIcon-avatar-1 CharIcon-brown-18"
          classes={
            Object {
              "colorDefault": "MuiAvatar-colorDefault-22",
              "img": "MuiAvatar-img-23",
              "root": "MuiAvatar-root-21",
            }
          }
          component="div"
        >
          <div
            className="MuiAvatar-root-21 MuiAvatar-colorDefault-22 CharIcon-avatar-1 CharIcon-brown-18"
          >
            o
          </div>
        </Avatar>
      </WithStyles(Avatar)>
    </CharIcon>
  </WithStyles(CharIcon)>
</MockRouter>
`;
